### PR TITLE
M4028 As dataset/entity owner I want to edit my entity metadata via the UI to keep up with changes with my users

### PR DESCRIPTION
**Pull request by Mark-de-Haan**
#### Checklist
- [ ] Functionality works & meets specifications
- [ ] Code reviewed
- [ ] Code unit/integration/system tested
- [ ] User documentation updated
- [ ] (If you have changed REST API interface) view-swagger.ftl updated
- [ ] Test plan template updated
- [ ] Clean commits

### CODE REVIEW

**Comment by ConnorStroomberg** [comment]
When i open the metadata manager plugin for the first time is see a black screen for the most part, maybe it is nice to select the first entity by default.

**Comment by ConnorStroomberg** [comment]
When is select the entity named 'Static content' form the entity select and press the delete entity button, and then select 'yes delete' i get two errors flying into the screen, one with the message: 'unknown errror', the other warning me i can not delete the system entity. 

I would the better to disable deleting for system entities, and have some kind of indicator a entity being viewed is a system entity.

**Comment by ConnorStroomberg** [comment]
The entity manager show a set of buttons and fields, the save button 'save all changes' has text on in, the delete entity button has an icon and a label, this inconsistency seems confusing to me.

**Comment by ConnorStroomberg** [comment]
When i select the entity 'Static content' and update a field ( for instance change the name of a attribute) i can not save the change ( i cannot update a system entity). Read only entities ( or  fields of entities ) should be show readonly.

**Comment by ConnorStroomberg** [comment]
When i select the plus icon the 'label' and 'description' fields are already filled out ( while the select field uses placeholder functionality). Use of placeholder text is needed if text in not a sensible default.

**Comment by ConnorStroomberg** [comment]
It is not clear to me when removing a attribute from the attribute list is this change is persisted to the backend or if this is done on save.

**Comment by ConnorStroomberg** [comment]
Then i create a new entity the first thing i see is a delete button then i press this , a get an error

**Comment by ConnorStroomberg** [comment]
When i create a new entity the value of the previously selected entity still remains in the entity select, this is confusing as now the name in the select has no link the the data on the screen.

**Comment by ConnorStroomberg** [comment]
When i create a new entity and save the entity nothing happens ( nothing seems to happen) when i press save again i get an error.

**Comment by ConnorStroomberg** [comment]
The entities in the entity select do not appear to be sorted in any particular other, alphabetically would be nice.

**Comment by ConnorStroomberg** [comment]
When i create an new entity i can not find it in the entity select.

**Comment by ConnorStroomberg** [comment]
when i delete my newly created entity i get and error

**Comment by ConnorStroomberg** [comment]
Probably a 'test plan' is needed

File: docs/SUMMARY.md [hunk 1]
```diff
@@ -27,6 +27,7 @@
         - [REST api V2](developer_documentation/ref-rest2.md)
         - [Swagger specification](developer_documentation/ref-swagger.md)
     - [FAIR](developer_documentation/guide-fair.md)
+    - [Frontend development](developer_documentation/frontend-development.md)
```
Comment (ConnorStroomberg) [inline docs/SUMMARY.md]: move to separate commit

File: docs/developer_documentation/frontend-development.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
+# Guide to developing MOLGENIS Frontend
```
Comment (ConnorStroomberg) [inline docs/developer_documentation/frontend-development.md]: move to seperate commit

File: intellij-molgenis-style.xml [hunk 3]
```diff
@@ -1,5 +1,14 @@
 <code_scheme name="molgenis">
```
Comment (ConnorStroomberg) [inline intellij-molgenis-style.xml]: move to separate PR

File: docs/developer_documentation/frontend-development.md (hunk 2 above)
Comment (Mark-de-Haan) [inline docs/developer_documentation/frontend-development.md]: why? We are going to squash it right?

File: molgenis-metadata-manager/src/main/frontend/README.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# frontend
+
+> MOLGENIS Metadata editor
+
+## Build Setup
+
+``` bash
+# install dependencies
+npm install
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/README.md]: should this be changed to yarn commands ?

Comment (Mark-de-Haan) [inline molgenis-metadata-manager/src/main/frontend/README.md]: Yes

File: molgenis-metadata-manager/src/main/frontend/index.html [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  <head>
+    <meta charset="utf-8">
+    <title>frontend</title>
+  </head>
+  <body>
+    <div id="app"></div>
+    <script>
+      window.__INITIAL_STATE__ = {
+        baseUrl: '/',
+        lng: 'en',
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/index.html]: why 'lng' instead of 'language' ?

Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/index.html]: why is the set in the template instead of the app code ?

File: molgenis-metadata-manager/src/main/frontend/package.json [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+  "dependencies": {
+    "@molgenis/molgenis-i18n-js": "^0.0.5",
+    "@molgenis/molgenis-api-client": "^1.0.0",
+    "bootstrap": "4.0.0-alpha.6",
+    "flow": "^0.2.3",
+    "flow-babel-webpack-plugin": "^1.0.1",
+    "font-awesome": "^4.7.0",
+    "v-toaster": "^1.0.2",
+    "vue": "^2.2.6",
+    "vue-multiselect": "next",
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/package.json]: 'next' seems a bit wild ... ?

File: molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerAttributeEditForm.vue [hunk 7]
```diff
@@ -0,0 +1,456 @@
...
+
+          <div class="form-group row">
+            <label class="col-3 col-form-label">Label</label>
+            <div class="col">
+              <input v-model="label" class="form-control" type="text">
+            </div>
+          </div>
+
+          <div class="form-group row">
+            <label class="col-3 col-form-label">Description</label>
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerAttributeEditForm.vue]: missing 'for' attribute for labels

File: molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerAttributeEditForm.vue [hunk 8]
```diff
@@ -0,0 +1,456 @@
...
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<style>
+  /*screen-md border on inner column when columns aligned next to each other*/
+  @media (min-width: 768px) {
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerAttributeEditForm.vue]: replace with bootstrap mixins

File: molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerAttributeEditForm.vue [hunk 9]
```diff
@@ -0,0 +1,456 @@
...
+    .col-md-3.attribute-tree {
+      border-right: solid black thin;
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerAttributeEditForm.vue]: black is hardcoded style, what if we want to have a 'dark' theme ?

File: molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerAttributeEditForm.vue [hunk 10]
```diff
@@ -0,0 +1,456 @@
...
+  import { CREATE_ATTRIBUTE } from '../store/actions'
+  import { getConfirmBeforeDeletingProperties } from '../store/getters'
+
+  import Multiselect from 'vue-multiselect'
+
+  export default {
+    name: 'metadata-manager-attribute-edit-form',
+    methods: {
+      deleteAttribute (selectedAttribute) {
+        this.$swal(getConfirmBeforeDeletingProperties(selectedAttribute.label)).then(() => {
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerAttributeEditForm.vue]: where is $swal comming from here ?

File: molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerEntityEditForm.vue [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+
+  /*screen-md border on inner column when columns aligned next to each other*/
+  @include media-breakpoint-up(md) {
+    .col-md-4.inner-column {
+      border-right: solid black thin;
+    }
+  }
+
+  .multiselect__tag {
+    background-color: darken($red, 20%);
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerEntityEditForm.vue]: why red ? this seems to signal error or waning to me

File: molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerEntityEditForm.vue [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+  @import "~mixins";
+
+  .checkbox-column {
+    padding-top: 11px
+  }
+
+  /*screen-md border on inner column when columns aligned next to each other*/
+  @include media-breakpoint-up(md) {
+    .col-md-4.inner-column {
+      border-right: solid black thin;
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerEntityEditForm.vue]: back is a hardcoded theme style

File: molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerEntityEditForm.vue [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+      },
+      description: {
+        get () {
+          return this.$store.state.editorEntityType.description
+        },
+        set (value) {
+          this.$store.commit(UPDATE_EDITOR_ENTITY_TYPE, { key: 'description', value: value })
+        }
+      },
+      package0: {
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerEntityEditForm.vue]: why 'package0' ? this name seems strange to me

File: molgenis-metadata-manager/src/main/frontend/src/components/generic-components/AttributeTreeNode.vue [hunk 14]
```diff
@@ -0,0 +1,49 @@
+<template>
+  <!-- TODO: Buttons for changing order -->
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/components/generic-components/AttributeTreeNode.vue]: remove todo

File: molgenis-metadata-manager/src/main/frontend/src/components/generic-components/AttributeTreeNode.vue [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+<style lang="scss">
+  @import "~variables";
+  @import "~mixins";
+
+  li {
+    list-style-type: none;
+  }
+
+  li:hover {
+    cursor: pointer;
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/components/generic-components/AttributeTreeNode.vue]: maybe change background color as well

File: molgenis-metadata-manager/src/main/frontend/src/store/actions.js [hunk 16]
```diff
@@ -0,0 +1,192 @@
...
+export const CREATE_ATTRIBUTE = '__CREATE_ATTRIBUTE__'
+export const SAVE_EDITOR_ENTITY_TYPE = '__SAVE_EDITOR_ENTITY_TYPE__'
+export const GET_ATTRIBUTE_TYPES = '__GET_ATTRIBUTE_TYPES__'
+
+export default {
+  /**
+   * Retrieve all Packages and filter on non-system Packages
+   */
+  [GET_PACKAGES] ({commit}) {
+    // TODO filter system packages
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/store/actions.js]: This seems like a major change to functionality, not a minor todo

File: molgenis-metadata-manager/src/main/frontend/src/store/actions.js [hunk 17]
```diff
@@ -0,0 +1,192 @@
...
+      .then(response => {
+        // Never reached due to https://github.com/molgenis/molgenis-api-client/issues/1
+      }, error => {
+        if (error.errors) {
+          commit(CREATE_ALERT, {
+            type: 'error',
+            message: error.errors[0].message
+          })
+        } else {
+          // Clear selected editorEntityType
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/store/actions.js]: i don't think we need all of these code comments here , the const names seem clear to me

File: molgenis-metadata-manager/src/main/frontend/src/store/getters.js [hunk 18]
```diff
@@ -0,0 +1,74 @@
+/**
+ * Returns a property object used for creating a sweetalert2 modal when trying to delete an EntityType
+ */
+export const getConfirmBeforeDeletingProperties = (identifier) => {
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/store/getters.js]: why not export the object directly instead of using a getter function

File: molgenis-metadata-manager/src/main/frontend/src/store/getters.js [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+
+export default {
+  /**
+   * Returns the EntityType object based on the ID of the selected EntityType
+   */
+  getSelectedEntityType: state => state.entityTypes.find(entityType => entityType.id === state.selectedEntityTypeId),
+  /**
+   * Returns the Attribute object based on the ID of the selected Attribute
+   */
+  getSelectedAttribute: state => state.editorEntityType && state.editorEntityType.attributes.find(attribute => attribute.id === state.selectedAttributeId),
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/store/getters.js]: these very long lines , are hard to read, debug or comment on.

File: molgenis-metadata-manager/src/main/frontend/src/store/mutations.js [hunk 20]
```diff
@@ -0,0 +1,91 @@
+export const SET_PACKAGES = '__SET_PACKAGES__'
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/store/mutations.js]: missing flow type annotations on the params

File: molgenis-metadata-manager/src/main/frontend/src/store/state.js [hunk 21]
```diff
@@ -0,0 +1,28 @@
+// @flow
+export const INITIAL_STATE = window.__INITIAL_STATE__ || {}
+
+import type { Alert, Package, EditorEntityType } from './utils/flow.types'
+
+export type State = {
+  alert: Alert,
+  packages: Array<Package>,
+  entityTypes: Array<Object>, // TODO create EntityType and Attribute type in flow.types.js
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/store/state.js]: implement todo

File: molgenis-metadata-manager/src/main/frontend/src/store/utils/flow.types.js [hunk 22]
```diff
@@ -0,0 +1,98 @@
+// @flow
+export type Alert = {
+  'message': ?string,
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/store/utils/flow.types.js]: all properties are optional this seems stange to me

File: molgenis-metadata-manager/src/main/frontend/src/store/utils/flow.types.js [hunk 23]
```diff
@@ -0,0 +1,98 @@
...
+  'label': ?string
+}
+
+export type EditorEntityType = {
+  'id': string,
+  'label': ?string,
+  'i18nLabel': ?any,
+  'description': ?string,
+  'i18nDescription': ?any,
+  'abstract0': ?boolean,
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/store/utils/flow.types.js]: 'isAbstract' indicates a boolean and is not a reserved word

File: molgenis-metadata-manager/src/main/frontend/src/store/utils/flow.types.js [hunk 24]
```diff
@@ -0,0 +1,98 @@
...
+  'backend': ?string,
+  'package0': ?EditorPackageIdentifier,
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/store/utils/flow.types.js]: 'packageIdentifier' would maybe be clearer

File: molgenis-metadata-manager/src/main/frontend/src/store/utils/utils.js [hunk 25]
```diff
@@ -0,0 +1,68 @@
+// @flow
+import type { EditorEntityType, EditorAttribute } from './flow.types'
+
+export const toEntityType = (editorEntityType: Object): EditorEntityType => {
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/store/utils/utils.js]: move to action as these seem like private methods

File: molgenis-metadata-manager/src/main/frontend/src/store/utils/flow.types.js [hunk 26]
```diff
@@ -0,0 +1,98 @@
+// @flow
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/store/utils/flow.types.js]: why is this file located in store/util ? I would think it would be top level in frondend

File: molgenis-metadata-manager/src/main/frontend/test/unit/specs/utils.spec.js [hunk 27]
```diff
@@ -0,0 +1,85 @@
+import * as utils from 'store/utils/utils'
+
+describe('utilities', () => {
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/test/unit/specs/utils.spec.js]: tests use a lot of undefineds maybe include a more filled out type as well

File: molgenis-metadata-manager/src/main/java/org/molgenis/metadata/manager/mapper/AttributeMapper.java [hunk 28]
```diff
@@ -0,0 +1,222 @@
...
+import java.util.stream.IntStream;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toMap;
+import static java.util.stream.StreamSupport.stream;
+import static org.molgenis.data.i18n.LanguageService.getLanguageCodes;
+import static org.molgenis.data.support.AttributeUtils.getI18nAttributeName;
+
+@Component
+public class AttributeMapper
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/java/org/molgenis/metadata/manager/mapper/AttributeMapper.java]: add javadoc explaining the function of the class

File: molgenis-metadata-manager/src/main/java/org/molgenis/metadata/manager/mapper/AttributeMapper.java [hunk 29]
```diff
@@ -0,0 +1,222 @@
...
+		String defaultValue = attribute.getDefaultValue();
+		Integer sequenceNumber = attribute.getSequenceNumber() != null ? attribute.getSequenceNumber() : 0;
+		return EditorAttribute
+				.create(id, name, type, parent, refEntityType, mappedByEntityType, orderBy, expression, nullable, auto,
+						visible, label, i18nLabel, description, i18nDescription, aggregatable, enumOptions, rangeMin,
+						rangeMax, readonly, unique, tags, visibleExpression, validationExpression, defaultValue,
+						sequenceNumber);
+	}
+
+	private ImmutableMap<String, String> toI18nLabel(Attribute attribute)
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/java/org/molgenis/metadata/manager/mapper/AttributeMapper.java]: these toI18n functions do not seem part of the metadata editor and could/should probably be moved to a separate service.

File: molgenis-metadata-manager/src/test/java/org/molgenis/metadata/manager/mapper/EntityTypeMapperTest.java [hunk 30]
```diff
@@ -0,0 +1,242 @@
...
+		EntityType parentEntityType = mock(EntityType.class);
+		when(entityTypeParentMapper.toEntityTypeReference(editorEntityTypeParent)).thenReturn(parentEntityType);
+		@SuppressWarnings("unchecked")
+		Iterable<Tag> tags = mock(Iterable.class);
+		when(tagMapper.toTagReferences(editorTags)).thenReturn(tags);
+		@SuppressWarnings("unchecked")
+		Iterable<Attribute> attributes = mock(Iterable.class);
+		when(attributeMapper.toAttributes(editorAttributes, editorEntityType)).thenReturn(attributes);
+		assertEquals(entityType, entityTypeMapper.toEntityType(editorEntityType));
+		// FIXME id label lookup attribute modifications
```
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/test/java/org/molgenis/metadata/manager/mapper/EntityTypeMapperTest.java]: fix fixme :)

File: molgenis-navigator/src/main/frontend/src/App.vue [hunk 31]
```diff
@@ -11,19 +11,8 @@
   </div>
 </template>
 
-<style>
```
Comment (ConnorStroomberg) [inline molgenis-navigator/src/main/frontend/src/App.vue]: Move to other pull request

File: molgenis-navigator/src/main/frontend/test/unit/coverage/cobertura/cobertura-coverage.xml [hunk 32]
```diff
@@ -1,301 +0,0 @@
-<?xml version="1.0" ?>
```
Comment (ConnorStroomberg) [inline molgenis-navigator/src/main/frontend/test/unit/coverage/cobertura/cobertura-coverage.xml]: move to other PR

File: molgenis-navigator/src/main/frontend/test/unit/specs/MgApp.spec.js [hunk 33]
```diff
@@ -1,7 +0,0 @@
-import MgApp from 'MgApp.vue'
```
Comment (ConnorStroomberg) [inline molgenis-navigator/src/main/frontend/test/unit/specs/MgApp.spec.js]: move to other pll request

File: molgenis-metadata-manager/src/main/frontend/package.json (hunk 6 above)
Comment (Mark-de-Haan) [inline molgenis-metadata-manager/src/main/frontend/package.json]: Normal version numbers or not yet compatible with Vue2, this is the only option we get.

File: molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerAttributeEditForm.vue (hunk 10 above)
Comment (Mark-de-Haan) [inline molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerAttributeEditForm.vue]: in main.js:
```js
import VueSweetAlert from 'vue-sweetalert'
Vue.use(VueSweetAlert)
```

This kind if inclusion exposes sweet alert in all child components.
Similar to vuex, where you do `this.$store...`

File: molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerEntityEditForm.vue (hunk 13 above)
Comment (Mark-de-Haan) [inline molgenis-metadata-manager/src/main/frontend/src/components/MetadataManagerEntityEditForm.vue]: This is the model as it is coming from the server, didn't want to change it due to collision errors on save

File: molgenis-metadata-manager/src/main/frontend/src/store/utils/flow.types.js (hunk 22 above)
Comment (Mark-de-Haan) [inline molgenis-metadata-manager/src/main/frontend/src/store/utils/flow.types.js]: You use the same mutation to set everything to null..
or is there another way to indicate it is allowed to be null?

File: molgenis-metadata-manager/src/main/frontend/src/store/utils/flow.types.js (hunk 23 above)
Comment (Mark-de-Haan) [inline molgenis-metadata-manager/src/main/frontend/src/store/utils/flow.types.js]: Keeping consistent with model coming from the server

File: molgenis-metadata-manager/src/main/frontend/src/store/utils/flow.types.js (hunk 24 above)
Comment (Mark-de-Haan) [inline molgenis-metadata-manager/src/main/frontend/src/store/utils/flow.types.js]: Keeping consistent with model coming from the server

File: molgenis-metadata-manager/src/main/frontend/src/store/getters.js (hunk 19 above)
Comment (Mark-de-Haan) [inline molgenis-metadata-manager/src/main/frontend/src/store/getters.js]: Case of getting used to lambda's I think,
else it would be
```js
getSelectedAttribute: function (state) {
    if (state.editorEntityType) {
       return  state.editorEntityType.attributes.find(function (attribute) {
            return attribute.id === state.selectedAttributeId
        }
    } else {
        return null
    }
}
```

File: docs/developer_documentation/frontend-development.md (hunk 2 above)
Comment (ConnorStroomberg) [inline docs/developer_documentation/frontend-development.md]: Its not part of this issue, if we keep adding stuff the PR is never going to get merged. Its best the have the small PRs

File: molgenis-metadata-manager/src/main/frontend/package.json (hunk 6 above)
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/package.json]: but should we be using it then ? why is there non stable version

File: molgenis-metadata-manager/src/main/frontend/src/store/getters.js (hunk 19 above)
Comment (ConnorStroomberg) [inline molgenis-metadata-manager/src/main/frontend/src/store/getters.js]: no its a case of using new line , i think

File: molgenis-metadata-manager/src/main/frontend/package.json (hunk 6 above)
Comment (Mark-de-Haan) [inline molgenis-metadata-manager/src/main/frontend/package.json]: Its stable, the developer is just being an idiot with versioning...

File: molgenis-metadata-manager/src/main/java/org/molgenis/metadata/manager/mapper/AttributeMapper.java (hunk 29 above)
Comment (Mark-de-Haan) [inline molgenis-metadata-manager/src/main/java/org/molgenis/metadata/manager/mapper/AttributeMapper.java]: i18n is part of the metadata of EntityTypes, so it should be here, but I don't really have an idea on how to use it though (also TODO/ out of scope in mockups)
